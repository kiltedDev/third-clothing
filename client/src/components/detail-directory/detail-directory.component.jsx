import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectContactDetailsForSharing } from '../../redux/contact/contact.selectors';

import ContactItem from '../contact-item/contact-item.component';

import './detail-directory.styles.scss';

const DetailDirectory = ({ contactDetails }) => (
  <div className='detail-directory'>
    {contactDetails.map(({ id, ...otherProps }) => (
      <ContactItem key={id} {...otherProps} />
    ))}
  </div>
);

const mapStateToProps = createStructuredSelector({
  contactDetails: selectContactDetailsForSharing
});

export default connect(mapStateToProps)(DetailDirectory);
