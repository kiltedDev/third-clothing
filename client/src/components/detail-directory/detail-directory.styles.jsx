import styled from 'styled-components';

export const DirectoryMenu = styled.div`
  width: auto;
  margin: auto;
  display: grid;
  grid-template-columns: 200px 200px 200px;
  flex-wrap: wrap;
  justify-content: space-between;
  @media screen and (max-width: 800px) {
    display: flex;
  }
`

DirectoryMenu.displayName = 'DirectoryMenu';
