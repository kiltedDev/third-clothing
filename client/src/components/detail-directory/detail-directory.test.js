import { shallow } from 'enzyme';
import React from 'react';
import { DetailDirectory } from './detail-directory.component';

describe('DetailDirectory', () => {
  let wrapper;

  beforeAll(() => {
    const mockProps = {
      contactDetails: [{
        id: 1,
        title: 'FaceSpace',
        contactInfo: 'cool-guy-82',
        linkTo: 'www.facespace.com/cool-guy-82',
        iconSize: '50px'
      }]
    }
    wrapper = shallow(<DetailDirectory { ...mockProps } />)
  })
  it('should render DetailDirectory component',() => {
    expect( wrapper.debug() ).toMatchSnapshot();
  })
})
