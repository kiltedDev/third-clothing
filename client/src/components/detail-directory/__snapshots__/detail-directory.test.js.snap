// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailDirectory should render DetailDirectory component 1`] = `
"<DirectoryMenu className=\\"detail-directory\\">
  <withRouter(ContactItem) title=\\"FaceSpace\\" contactInfo=\\"cool-guy-82\\" linkTo=\\"www.facespace.com/cool-guy-82\\" iconSize=\\"50px\\" />
</DirectoryMenu>"
`;
